test/extended/apiserver/webhook.go

```diff
@@ -0,0 +1,386 @@
+				},
+				ObjectSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{objectSelector: "true"},
+				},
+			},
+		},
+	}
+	_, err = client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Create(context.Background(), config, metav1.CreateOptions{})
+	e2e.ExpectNoError(err, "creating webhook config %s in namespace %s", configName, namespace)
+
+	removeWebhookConfiguration := func() {
+		err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Delete(context.Background(), configName, metav1.DeleteOptions{})
+		e2e.ExpectNoError(err, "deleting webhook config %s in namespace %s", configName, namespace)
+	}
+
+	var (
+		cm       *v1.ConfigMap
+		cmClient = client.CoreV1().ConfigMaps(namespace)
+	)
+
+	createResourceFn := func() error {
+		cm = &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: string(uuid.NewUUID()),
+				Labels: map[string]string{
+					objectSelector: "true",
+				},
+			},
+		}
+		_, err := cmClient.Create(context.Background(), cm, metav1.CreateOptions{})
+		return err
+	}
+	deleteResourceFn := func() {
+		err = cmClient.Delete(context.Background(), cm.GetName(), metav1.DeleteOptions{})
+		e2e.ExpectNoError(err, "deleting config map %s in namespace %s", cm.Name, namespace)
+	}
+
+	err = hitValidatingWebhook(createResourceFn, deleteResourceFn)
+	if err != nil {
+		removeWebhookConfiguration()
+		e2e.ExpectNoError(err, "calling validating admission webhook")
+	}
+	return removeWebhookConfiguration
+}
+
+// hitValidatingWebhook tries to create a resource and expect it to fail (deny)
+// it tries to send a few request before giving up because registering a webhook is not instant and might take up to a few seconds
+func hitValidatingWebhook(createResourceFn func() error, deleteResourceFn func()) error {
+	return wait.PollImmediate(100*time.Millisecond, 30*time.Second, func() (bool, error) {
+		err := createResourceFn()
+		if err != nil {
+			// the always-deny webhook does not provide a reason, so check for the error string we expect
+			if strings.Contains(err.Error(), "denied") {
+				return true, nil
+			}
+			return false, err
+		}
+		deleteResourceFn()
+		e2e.Logf("Calling webhook succeeded but it should fail trying one more time...")
+		return false, nil
+	})
+}
+
+func createRes(restConfig *rest.Config, mapper *restmapper.DeferredDiscoveryRESTMapper, gvr schema.GroupVersionResource, withWebhookObjectSelector bool, stub, namespace string) (dynamic.ResourceInterface, *unstructured.Unstructured, error) {
+	gvk, err := mapper.KindFor(gvr)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// supply a value for namespace if the scope requires
+	mapping, err := mapper.RESTMapping(gvk.GroupKind())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// ensure that any stub embedding the etcd test namespace
+	// is updated to use local test namespace instead.
+	stub = strings.Replace(stub, exetcd.TestNamespace, namespace, -1)
```

Comment:
some stubs embed `etcdstoragepathtestnamespace` namespace, for example 
```
Stub:             `{"clientName": "system:serviceaccount:etcdstoragepathtestnamespace:clientg", "metadata": {"name": "user:system:serviceaccount:etcdstoragepathtestnamespace:clientg"}, "scopes": ["user:info"], "userName": "user", "userUID": "cannot be empty"}`
```

this code makes sure that it will be replaced by the target namespace so that it can be found by the test.